libkbfs/dirty_bcache.go

```diff
@@ -98,25 +131,119 @@ func (d *DirtyBlockCacheStandard) IsDirty(
 		branch:   branch,
 	}
 
-	d.dirtyLock.Lock()
-	defer d.dirtyLock.Unlock()
-	_, isDirty = d.dirty[dirtyID]
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	_, isDirty = d.cache[dirtyID]
 	return
 }
 
-// DirtyBytesEstimate implements the DirtyBlockCache interface for
-// DirtyBlockCacheStandard.
-func (d *DirtyBlockCacheStandard) DirtyBytesEstimate() uint64 {
-	d.dirtyLock.Lock()
-	defer d.dirtyLock.Unlock()
-	if d.dirtyBytesEstimate == 0 {
-		// Users of this cache can update dirty blocks at will, so
-		// it's not possible to return the exact sizes of the blocks.
-		// Just cache what we have until we know for sure that it's
-		// changed (because a new block is added, for example).
-		for _, block := range d.dirty {
-			d.dirtyBytesEstimate += uint64(getCachedBlockSize(block))
+func (d *DirtyBlockCacheStandard) acceptNewWrite(newBytes int64) bool {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	// Accept any write, as long as we're not already over the limits.
+	canAccept := d.unsyncedDirtyBytes < d.maxSyncBufferSize &&
+		d.totalDirtyBytes < d.maxDirtyBufferSize
+	if canAccept {
+		d.unsyncedDirtyBytes += newBytes
+		d.totalDirtyBytes += newBytes
+	}
+	return canAccept
+}
+
+func (d *DirtyBlockCacheStandard) processPermission() {
```

Comment:
I'd like a comment explaining the logic of this loop, or more detailed comments in the code, since there's some state that carries across iterations and it's difficult to follow.